cg/models/orders/excel_sample.py

```diff
@@ -1,139 +1,147 @@
 from typing import List, Optional
-from pydantic.v1 import Field, validator
 
 from cg.constants.orderforms import REV_SEX_MAP, SOURCE_TYPES
 from cg.models.orders.sample_base import OrderSample
+from pydantic import AfterValidator, BeforeValidator, Field
+from typing_extensions import Annotated
+
+
+def parse_panels(value: str):
+    if not value:
+        return None
+    separator = ";" if ";" in value else None
+    if ":" in value:
+        separator = ":"
+    return value.split(separator)
+
+
+def validate_data_analysis(value):
+    data_analysis_alternatives = [
+        "Balsamic",  # OF 1508
+        "Balsamic QC",  # OF 1508
+        "Balsamic UMI",  # OF 1508
+        "fastq",  # OF 1605
+        "FLUFFY",  # OF 1604
+        "MicroSALT",  # OF 1603 (implicit)
+        "MIP DNA",  # OF 1508
+        "MIP RNA",  # OF 1508
+        "RNAfusion",  # OF 1508
+        "SARS-CoV-2",  # OF 2184
+        "No analysis",  # OF 1508, 1604, 2184
+    ]
+    if value not in data_analysis_alternatives:
+        raise AttributeError(f"'{value}' is not a valid data analysis")
+    return value
+
+
+def numeric_value(value: Optional[str]):
+    if not value:
+        return None
+    str_value = value.rsplit(".0")[0]
+    if str_value.replace(".", "").isnumeric():
+        return str_value
+    raise AttributeError(f"Order contains non-numeric value '{value}'")
+
+
+def validate_parent(value: str):
+    return None if value == "0.0" else value
+
+
+def validate_source(value: Optional[str]):
+    if value not in SOURCE_TYPES:
+        raise ValueError(f"'{value}' is not a valid source")
+    return value
+
+
+def convert_sex(value: Optional[str]):
+    if not value:
+        return None
+    value = value.strip()
+    return REV_SEX_MAP.get(value, "unknown")
+
+
+def convert_to_lower(value: Optional[str]):
+    return value.lower()
+
+
+def convert_to_priority(value: Optional[str]):
+    return "priority" if value == "förtur" else value
```

Comment:
```suggestion
    return "priority" if value == "förtur" else value
```

I think we would want an english named value here. Can we create a constant for "förtur" which is its english translation?

Reply:
I don't know if there is a functioning translator for arbitrary strings? And I think "förtur" is the only Swedish word that seems to be supported when filling out certain forms, allowed values are in the PriorityEnum, i.e. "research, standard, priority, express or clinical_trials". So if none of those are specified we throw an error (unless "förtur" is specified which is converted to priority). But a constant should be doable.